backend/common/utils/dataclass.py

```diff
@@ -0,0 +1,38 @@
+from dataclasses import asdict, is_dataclass
+
+
+def convert_dataclass_to_dict(data, remove_nones=True):
+    """
+    Convert a dataclass or dict of dataclasses to a dict and optionally
+    remove all keys with None values (default behavior).
+
+    Arguments
+    ---------
+    data - dataclass or dict of dataclasses
+    remove_nones - bool, if True, remove all keys with None values
+    """
+    if is_dataclass(data):
+        data = asdict(data)
+
+    elif isinstance(data, dict):
+        for key, value in data.items():
+            if is_dataclass(value):
+                data[key] = asdict(value)
+
+    return _remove_none_values(data)
```

Comment:
also the `remove_nones` param is not actually used here lol

Reply:
Whoopsie